### PR TITLE
bazel: moving out c-deps from Root BUILD

### PR DESCRIPTION
**Pull request by alan-mas**
This is one of the re-factoring that we are planning to do
and it is about to move c-deps out of root BUILD #56051

We are not only creating a new BUILD file but also changing all the other BUILDS
that are getting affected due to this dependency

Release note: None

### CODE REVIEW

**Comment by alan-mas** [comment]
cc @jlinder

File: pkg/sql/lex/BUILD.bazel [hunk 1]
```diff
@@ -6,6 +6,7 @@ go_library(
         "encode.go",
         "experimental_keywords.go",
         "keywords.go",  # keep
+        "reserved_keywords.go",
```
Comment (otan) [inline pkg/sql/lex/BUILD.bazel:9]: i think this needs to be rebased

File: c-deps/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    }),
+    lib_source = "@krb5//:all",
+    make_commands = [
+        "make",
+        "mkdir -p libkrb5/lib",
+        "cp libkrb5/libgssapi_krb5.a libkrb5/lib",
+    ],
+    static_libraries = ["libgssapi_krb5.a"],
+    visibility = ["//visibility:public"],
+)
```
Comment (otan) [inline c-deps/BUILD.bazel:113]: can we move the WORKSPACE items that point to the external repositories into a WORKSPACE file in this directory here?

basically all of https://github.com/cockroachdb/cockroach/blob/master/WORKSPACE#L63-L115.

**Comment by otan** [comment]
looking good! some minor comments ^

File: pkg/sql/lex/BUILD.bazel (hunk 1 above)
Comment (alan-mas) [reply pkg/sql/lex/BUILD.bazel:9]: Yes, you were right, I just rebase to the last commit in master:
e533f1939c55c34470e1c09958fe9f8205b63805

(at least by this time)

**Comment by otan** [comment]
can you fix the first line of your commit message to be "bazel: moving out c-deps from Root BUILD" -- thanks!

File: c-deps/BUILD.bazel (hunk 2 above)
Comment (alan-mas) [reply c-deps/BUILD.bazel:113]: As far as I know WORKSPACE file works as your ROOT directory for the complete project structure with Bazel (at least on my experience) you can have multiple WORKSPACES files so only the dependencies you need will be loaded:
https://docs.bazel.build/versions/master/external.html

Tbh I am not totally sure but I can give a try.

Should we do it in another pr?

File: pkg/sql/lex/BUILD.bazel (hunk 1 above)
Comment (irfansharif) [reply pkg/sql/lex/BUILD.bazel:9]: This diff should go away, I think. This file was moved to pkg/sql/lexbase. You might have a leftover generated file from an earlier commit. You should be able to `git clean -fx` it away.

Comment (alan-mas) [reply pkg/sql/lex/BUILD.bazel]: You were right, seemed to be an issue some garbage code that was there but I already fixed it.

Thank you for the git clean command as well

File: c-deps/BUILD.bazel (hunk 2 above)
Comment (otan) [reply c-deps/BUILD.bazel:113]: yeah would rather have the workspace in a separate directory.
ok, feel free to give it a shot in a separate PR.

**Review by otan — APPROVED** [review]

**Comment by irfansharif** [comment]
(Fixed a typo is all)

bors r+

**Comment by Amruta-Ranade** [comment]
@alan-mas Thank you for contributing to CockroachDB this year. As a token of our appreciation, we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) @Amruta so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please contact me as soon as possible :)

**Comment by Amruta-Ranade** [comment]
@alan-mas Thank you for contributing to CockroachDB this year. As a token of our appreciation, we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) @Amruta so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please contact me as soon as possible :)

**Comment by irfansharif** [comment]
Yo @alan-mas, I pre-emptively deleted Amruta's comment from earlier. I thought she was running a faulty script but I was mistaken. Copying over for posterity:

> Thank you for contributing to CockroachDB this year. As a token of our appreciation, we would like to send you a gift. Please DM me on our community Slack @amruta so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please contact me as soon as possible 

Definitely do reach out! I'm also curious what this gift is and how I can solicit one of my own.

**Pull request merged**